llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -5155,7 +5155,39 @@ struct StrictFPUpgradeVisitor : public InstVisitor<StrictFPUpgradeVisitor> {
 };
 } // namespace
 
-void llvm::UpgradeFunctionAttributes(Function &F) {
+static void
+CopyModuleAttributeToFunction(Function &F, StringRef FnAttrName,
+                              StringRef ModAttrName,
+                              std::pair<StringRef, StringRef> Values) {
+  Module *M = F.getParent();
+  if (!M)
+    return;
```

Comment:
Consider making this an assertion unless you observed cases where this was false?  The call chain checks this already in `llvm::UpgradeFunctionAttributes`, so perhaps an assertion here in the callee is more appropriate, in case this function grows more callers in the future.